packages/block-library/src/post-content/index.php

```diff
@@ -18,6 +18,10 @@ function render_block_core_post_content( $attributes, $content, $block ) {
 		return '';
 	}
 
+	if ( empty( get_the_content( $block->context['postId'] ) ) ) {
```

Comment:
A few lines below we're doing `get_the_content( null, false, $block->context['postId'] )`
Could we move this check below the `in_the_loop()` check, set the content as a variable and reuse it in line 33? That would avoid getting the content twice, and it would make sure that it runs after `the_post` so WooCommerce (and other plugins that filter `the_content` do their thing properly.